### DISCUSSION TITLE
Adjust Delly filtering parameters

### DISCUSSION THREAD

**Discussion by maotian06** [uclahs-cds/pipeline-call-sSV#115] [Q&A]
Hi @Faizal-Eeman @tyamaguchi-ucla @WuSelina 

I was wondering if you have any suggestions for adjusting the Delly parameters for high-coverage samples. Thanks!!!

```
    /**
    * Set up the Delly filtering parameters
    * The below default values are recommended to reduce runtimes.
    * See - https://github.com/dellytools/delly 'Delly is running too slowly what can I do?' for more
    */
    map_qual = 20
    min_clique_size = 5
    mad_cutoff = 15
```

**Comment by Faizal-Eeman** [comment]
@maotian06 

The default parameters can be considered on the lower limits of stringent filters, so you should generally observe all high quality SVs from high coverage data.

May I know what you are trying to achieve by adjusting the parameters?

**Comment by maotian06** [comment]
Here is an example of my samples in which I observed too many SV events.

![image](https://github.com/uclahs-cds/pipeline-call-sSV/assets/47042983/e3961482-83a1-4e69-9702-4bead3f2090b)

When discussed with @tyamaguchi-ucla , he pointed out this might be due to the default Delly parameter was set for coverage ~ 30x. So I was wondering how should I adjust the parameter to fit high coverage data? I have samples from 60~90X.

**Reply by tnyamaguchi** [reply]
@maotian06 

>When discussed with @tyamaguchi-ucla , he pointed out this might be due to the default Delly parameter was set for coverage ~ 30x. 

This is not exactly what I meant. I tried to highlight that the thresholds need to be adjusted for coverage. The numbers are just examples. You may also want to consider adjusting the thresholds for purity and clonality as well. You should be able to learn the baseline by looking at the VCF outputs (Split reads, PE support numbers).

**Reply by tnyamaguchi** [reply]
>You should be able to learn the baseline by looking at the VCF outputs (Split reads, PE support numbers).

If I recall, @WuSelina has a script to visualize this.

**Comment by WuSelina** [comment]
Hey @maotian06, I did a couple things which resulted in many fewer SVs.

1) I looked at consensus SVs between DELLY and Manta using SURVIVOR, which I posted as a discussion:  (https://github.com/uclahs-cds/pipeline-call-sSV/discussions/100). This drastically cut the number of SVs, as Manta uses higher thresholds and thus returns notably fewer SVs compared to DELLY.

2) I increased the `min_clique_size` to 8, since my average sample coverage was 160x. This parameter specifies the min number of paired end and split read alignments required, so if I set it to 8, that would mean 5% (8 reads / 160 reads) of the reads have to support the SV (as I understand). 

Note: I looked at the consensus SVs (drastically fewer SVs because of Manta) before adjusting `min_clique_size`. If I recall correctly, just using DELLY, I would increase the `min_clique_size` threshold even higher (perhaps 10-15% of reads requirement).

**Comment by Faizal-Eeman** [comment]
When you increase the default parameter values, you will be compromising by loosing any potential SVs which have lower reads or MAPQ support which would be ignored by DELLY. This would also mean you have to re-run the pipeline.


However, if you are only interested in reducing the SVs or only want High Quality SVs then you can filter the Delly output to consider SVs with

- Coverage >30X (30X minimum support for an SV call would suffice to indicate confidence)
- Clique size or PE support for the SV should be >= 15X 
- Mapping Quality > 30, but you can consider MAPQ >= 60 for an even stringent filtering


Also take a look [here](https://github.com/uclahs-cds/pipeline-call-sSV#nextflow-config-file-parameters) for descriptions of clique_size and mad_cutoff. Looks for these fields in the VCF and filter them downstream.

**Reply by Faizal-Eeman** [reply]
> * Coverage >30X (30X minimum support for an SV call would suffice to indicate confidence)
> * Clique size or PE support for the SV should be >= 15X
> * Mapping Quality > 30, but you can consider MAPQ >= 60 for an even stringent filtering

These are generally the industry thresholds considered for curating variants before an orthogonal confirmation.

**Reply by tnyamaguchi** [reply]
>> - Coverage >30X (30X minimum support for an SV call would suffice to indicate confidence)
>> - Clique size or PE support for the SV should be >= 15X
>> - Mapping Quality > 30, but you can consider MAPQ >= 60 for an even stringent filtering

>These are generally the industry thresholds considered for curating variants before an orthogonal confirmation.

@Faizal-Eeman do you have any reference for these recommendations?

**Reply by maotian06** [reply]
Thanks @WuSelina @Faizal-Eeman @tyamaguchi-ucla 
I will test it out by directly filtering Delly output and probably re-run Delly in one or two samples first. I will post those results here once it is done.
As @tyamaguchi-ucla suggested, I have consulted Helen for consensus SVs and posted her suggestion in #100

**Comment by raagagrawal** [comment]
This is a question I am asking myself recently. @maotian06 did you ever settle on a setting that works for you? I have been reading around this issue and can't find anyone rigorously testing clique size by coverage for example. The setting of 5 seems to derive from the package author scaling up the initial default of 2, and not on an empirical finidng.
